pipeline-model-definition/src/test/resources/postStage/postFailureSuccessInParallel.groovy

```diff
@@ -0,0 +1,109 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+pipeline {
+    agent none
+    stages {
+        stage('Parallel') {
+            parallel {
+                stage('Child 1') {
+                    agent any
+                    steps { error 'Fail in Child 1' }
+                    post {
+                        failure {
+                            echo "Found failure in Child 1"
+                        }
+                        success {
+                            echo "Found success in Child 1"
+                        }
+                    }
+                }
+                stage('Child 2') {
+                    agent any
+                    stages {
+                        stage('Nested 1') {
+                            steps {
+                                echo 'Nested 1'
+                            }
+                            post {
+                                always {
+                                    echo "Post Nested 1 ran"
+                                }
+                            }
+                        }
+                        stage('Nested 2') {
+                            steps {
+                                sleep 2
+                                echo 'Nested 2'
+                            }
+                            post {
+                                always {
+                                    echo "Post Nested 2 ran"
+                                }
+                                success {
+                                    echo "Found success in Nested 2"
+                                }
+                                failure {
+                                    echo "Found failure in Nested 2"
+                                }
+                            }
+                        }
+                    }
+                    post {
+                        always {
+                            echo "Post Child 2 ran"
+                        }
+                        success {
+                            echo "Found success in Child 2"
+                        }
+                        failure {
+                            echo "Found failure in Child 2"
+                        }
+                    }
+                }
+                stage('Child 3') {
+                    steps {
+                        echo "Child 3"
+                    }
+                    post {
+                        always {
+                            error "Fail Child 3"
+                        }
+                        success {
+                            echo "Found success in Child 3"
+                        }
+                        failure {
+                            echo "Found failure in Child 3"
+                        }
+                    }
+                }
+            }
+            post {
```

Comment:
Would there be any value in having `success` and `failure` here too? With `failure` in the `logContains` part of the test, and `success` in the `logNotContains` part of the test.

Reply:
To verify that the top-most stuff in`post` still behaves as expected.

Reply:
Couldn't hurt - in practice, that's actually covered by a pile o' other tests, but hey, nothing wrong with testing it here too.

Reply:
:+1: . My thinking was, "make sure we still process these `success` and `failure` blocks after we've failed inside of the `parallel`. I'm sure that's being done elsewhere, but for cases like directly setting `currentBuild.result = 'FAILURE'` , I wasn't certain.